samza-core/src/main/java/org/apache/samza/zk/ZkUtils.java

```diff
@@ -79,11 +98,12 @@ public void connect() throws ZkInterruptedException {
     }
   }
 
-  public static ZkConnection createZkConnection(String zkConnectString, int sessionTimeoutMs) {
-    return new ZkConnection(zkConnectString, sessionTimeoutMs);
+  // reset all zk-session specific state
+  public void unregister() {
+    ephemeralPath = null;
   }
 
-  ZkClient getZkClient() {
+  public ZkClient getZkClient() {
```

Comment:
although it is a right thing to do, unfortunately we have some tests in a different package using it.
And I don't know an immediate way to change it.